File: SpriteCompression/src/Sprite.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Drawing;

namespace SpriteCompression
{
    public class Sprite
    {
        public int Width { get; private set; }
        public int Height { get; private set; }
        public int[] Pixels { get; private set; }

        public Sprite(string path)
        {
            Bitmap bitmap = new Bitmap(path);
            Width = bitmap.Width;
            Height = bitmap.Height;
            Pixels = new int[Width * Height];
            for (int y = 0; y < Height; y++)
            {
                for (int x = 0; x < Width; x++)
                {
                    Color pixel = bitmap.GetPixel(x, y);
                    Pixels[x + y * Width] = pixel.A << 24 | pixel.R << 16 | pixel.B << 8 | pixel.B;
                }
            }
        }

        public Sprite(BinaryReader br)
        {
            Width = 400;
            Height = 400;
            Pixels = new int[Width * Height];
            for (int i = 0; i < 400 * 400; i++)
                Pixels[i] = br.ReadInt32();
        }
    }
}

File: Compressor/src/Sprite.cpp
#include "Sprite.h"

#define STB_IMAGE_IMPLEMENTATION
#include <stb_image.h>

Sprite::Sprite(const String& path)
{
	int c;
	int w, h;
	pixels = stbi_load(path.c_str(), &w, &h, &c, STBI_rgb_alpha);
	width = (uint)w;
	height = (uint)h;
}